### DISCUSSION TITLE
How do i can create my custom placeholder?

### DISCUSSION THREAD

**Discussion by viet-innomize** [ueberdosis/tiptap#5448] [General Questions]
Example my usecase, i start create node, i select tag h1 ( h2 , h3, etc...), my custom placeholder will appear, after that when i typing to create note content, placeholder is remove !

**Comment by nperez0111** [comment]
You are describing how the placeholder extension is supposed to work

On empty nodes, the placeholder is set on the element as an attribute and shown via CSS.

**Reply by viet-innomize** [reply]
thanks, your reply ! I can create  some placeholder with placeholder extension, but it's seem to be difficult to custom style or add more content example add a tag with style default to content of placeholder. Can i use transaction of editor or something ... to create this at custom node. I don't understand how to use transaction  to handle some logic, i used this and loop... happening cratch app.

**Reply by nperez0111** [reply]
I'm sorry but this is not clear at all. Could you maybe rephrase this or use a translator maybe?

**Reply by viet-innomize** [reply]
sorry, my use case is "i want to custom placeholder like  "
![Uploading Screenshot 2024-08-06 at 17.00.16.png…]()

**Reply by nperez0111** [reply]
The screenshot did not upload
